app/controllers/providers_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+module Authenticators
+  module Oidc
+    class ProvidersController < RestController
+      include FindResource
+      include AssumedRole
+      include CurrentUser
+
+      def index
+        # Rails 5 requires parameters to be explicitly permitted before converting
+        # to Hash.  See: https://stackoverflow.com/a/46029524
+        allowed_params = %i[account]
+        
+        render(
+          json: Contexts::Authenticators::AvailableAuthenticators.new.call(
+            role: assumed_role(query_role),
+            account: params[:account]
+          )
+        )
+        
+      end
+
+      # The v5 API currently sends +acting_as+ when listing resources
+      # for a role other than the current user.
+      def query_role
+        params[:role].presence || params[:acting_as].presence
```

Comment:
This method feels a little soft from a functional perspective. What do you think about having it focus on resolving the role of the requestor (either defined or "acting as").  If we shift `assumed_role` into this method:

```suggestion
      def role
        assumed_role(params[:role].presence || params[:acting_as].presence)
      end
```
we have method which handles the resolution of the role depending on the provided parameters.